src/OrleansAzureUtils/Providers/Lease/AzureBlobLeaseProvider.cs

```diff
@@ -0,0 +1,140 @@
+﻿using Microsoft.WindowsAzure.Storage;
+using Microsoft.WindowsAzure.Storage.Blob;
+using Orleans.Runtime.Configuration;
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Runtime.CompilerServices;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Orleans.LeaseProviders
+{
+    public class AzureBlobLeaseProviderConfig
+    {
+        public string DataConnectionString { get; set; }
+        public string BlobContainerName { get; set; }
+    }
+    public class AzureBlobLeaseProvider : ILeaseProvider
+    {
+        private CloudBlobContainer container;
+        private AzureBlobLeaseProviderConfig providerConfig;
+        private CloudBlobClient blobClient;
+        public AzureBlobLeaseProvider(AzureBlobLeaseProviderConfig config)
+        {
+            var account = CloudStorageAccount.Parse(config.DataConnectionString);
+            this.blobClient = account.CreateCloudBlobClient();
+            this.providerConfig = config;
+        }
+
+        private async Task InitContainerIfNotExistsAsync()
+        {
+            if (this.container == null)
+            {
+                this.container = blobClient.GetContainerReference(this.providerConfig.BlobContainerName);
+                await this.container.CreateIfNotExistsAsync().ConfigureAwait(false); 
```

Comment:
If this call fails then it can never be retried. Seems like we should only assign `this.container` after this has completed successfully